android/src/main/java/com/RNAppleAuthentication/AppleAuthenticationAndroidModule.java

```diff
@@ -0,0 +1,216 @@
+package com.RNAppleAuthentication;
+
+import android.app.Activity;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentActivity;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+
+import com.RNAppleAuthentication.SignInWithAppleCallback;
+import com.RNAppleAuthentication.SignInWithAppleConfiguration;
+import com.RNAppleAuthentication.SignInWithAppleService;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.security.MessageDigest;
+
+public class AppleAuthenticationAndroidModule extends ReactContextBaseJavaModule {
+    private static final String E_NOT_CONFIGURED_ERROR = "E_NOT_CONFIGURED_ERROR";
+    private static final String E_SIGNIN_FAILED_ERROR = "E_SIGNIN_FAILED_ERROR";
+    private static final String E_SIGNIN_CANCELLED_ERROR = "E_SIGNIN_CANCELLED_ERROR";
+
+    private final ReactApplicationContext reactContext;
+
+    private @Nullable
+    SignInWithAppleConfiguration configuration;
+
+    public AppleAuthenticationAndroidModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
+    }
+
+    @Override
+    public String getName() {
+        return "RNAppleAuthModuleAndroid";
+    }
+
+    private static String bytesToHex(byte[] hash) {
+        StringBuffer hexString = new StringBuffer();
+        for (int i = 0; i < hash.length; i++) {
+        String hex = Integer.toHexString(0xff & hash[i]);
+        if(hex.length() == 1) hexString.append('0');
+            hexString.append(hex);
+        }
```

Comment:
This alternate style from probably the same stackoverflow https://stackoverflow.com/questions/2817752/java-code-to-convert-byte-to-hexadecimal avoids the padding issues and sign issues while still being pretty quick

```suggestion
        for (int i=0; i < hash.length; i++){
            hexString.append(Character.forDigit((hash[i] >> 4) & 0xF, 16));
            hextString.append(Character.forDigit((hash[i] & 0xF), 16));
        }
```